### PR TITLE
Rename `StateChannel.sol`

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
Proposed new names:

- `AppDefinition`: a contract with `applyAction, resolve, getTurnTaker, isStateTerminal`
- `AppInstance`: a contract that holds app state storage in a dispute. The `AppInstance.sol` contract also happens to contain framework code for dispute resolution
- "app": either an `AppDefinition` or `AppInstance` depending on context

### CODE REVIEW

File: utils/multisig.ts [hunk 1]
```diff
@@ -85,6 +85,9 @@ export class Multisig {
     args: any[],
     signers: ethers.Wallet[]
   ): Promise<ethers.types.Transaction> {
+    if (toContract.interface.functions[funcName] === undefined) {
+      throw new Error(`Tried to execute delegateCall to ${funcName} but no such function exists on the target contract`)
```
Comment (snario) [inline utils/multisig.ts]: The docstring & this comment should be consistent in using the word call vs delegatecall.

File: contracts/AppInstance.sol [hunk 2]
```diff
@@ -6,12 +6,12 @@ import "./lib/StaticCall.sol";
 import "./lib/Transfer.sol";
 
 
-/// @title StateChannel - A generalized state channel application contract
+/// @title AppInstance - A contract that defines a state channel application
```
Comment (cf19drofxots) [inline contracts/AppInstance.sol]: So by this terminology, deploying a state channel application gives an instance of an `AppInstance`?

Why not just name it `App` then? Then it's more intuitive to have an _instance_ of an app?

File: contracts/delegateTargets/ConditionalTransfer.sol [hunk 3]
```diff
@@ -26,17 +26,17 @@ contract ConditionalTransfer is Conditional {
     tx.execute();
   }
 
-  /// @notice Execute a fund transfer for a state channel in a finalized state
+  /// @notice Execute a fund transfer for a state channel app in a finalized state
   /// @param key The key in the nonce registry
   /// @param expectedNonce The expected nonce in the nonce registry
-  /// @param channelCfAddress Counterfactual address of the state channel contract
+  /// @param appCfAddress Counterfactual address of the state channel contract
```
Comment (cf19drofxots) [inline contracts/delegateTargets/ConditionalTransfer.sol]: `...of the app's contract`?

File: contracts/AppInstance.sol (hunk 2 above)
Comment (mvanderh) [reply contracts/AppInstance.sol]: I think it's to distinguish between ambiguous meanings for "App": Instance and Definition

Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply contracts/AppInstance.sol]: Yeap what @mvanderh said, I feel like having separate `App` and a `AppDefinition` contracts is more confusing

Comment (cf19drofxots) [reply contracts/AppInstance.sol]: Fair enough

**Comment by mvanderh** [comment]
I like this a lot! Big improvement on earlier naming.

Random idea: Maybe we should name the "AppInstance" something with "Dispute" in it? Since it's only ever used to resolve disputes on chain that would make sense to me.

**Pull request merged**